sharktank/sharktank/models/llama4/testing.py

```diff
@@ -0,0 +1,172 @@
+# Copyright 2025 Advanced Micro Devices, Inc
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+from ...layers import BaseCausalLMModel, LlamaModelConfig, LlamaHParams
+from ...types import Theta, unbox_tensor
+from transformers.models.llama4 import Llama4TextConfig
+
+import torch
+
+
+def config_to_hugging_face_text_config(config: LlamaModelConfig) -> Llama4TextConfig:
+    moe_layers = None
+    if config.hp.interleave_moe_layer_step is None:
+        moe_layers = config.moe_layers
+    rope_scaling = {
+        "factor": 8.0,
+        "high_freq_factor": 4.0,
+        "low_freq_factor": 1.0,
+        "original_max_position_embeddings": 8192,
+        "rope_type": "llama3",
+    }
```

Comment:
It is strange that this says `rope_type` is `llama3` but our config has `rope_scaling_type` `llama4`.
I forgot why Llama4 needed a separate RoPE path.

Reply:
I have changed it to `llama3` in our config to keep it consistent with the Transformers.